libr/core/canal.c

```diff
@@ -4040,6 +4041,12 @@ R_API void r_core_anal_esil(RCore *core, const char *str, const char *target) {
 				r_flag_space_set (core->flags, "syscalls");
 				int snv = (int)r_reg_getv (core->anal->reg, sn);
 				if (snv > 0) {
+
+					if (!strcmp (asmarch, "arm") && core->anal->bits == 16) {
+						//ut64?? todo downcast ok?
```

Comment:
i dont think we want to have this to be thumnb-specific. syscall number and syscall vector index are different things

Reply:
Still not sure what exercises this path so I could be wrong, But it seems like any time r_syscall_get gets called for thumb it should be with the syscall number?

Reply:
Syscalls are identified by the swi and the number, the swi is 0x80 in the case of linux , 0x21 in dos, etc.. the value in AX is the number of the syscall and the SWI is the vector table selection index.

So, saying this, i dont see why ARM/Thumb should be different than any other arch to handle this, just honor both values instead of using the default -1

Reply:
Thats not me calling -1 for arm, thats literally everywhere in the existing codebase. Im the only one not passing -1.